libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -150,12 +152,43 @@ string IRGenerator::generateFunction(FunctionDefinition const& _function)
 			retInit += generateInitialAssignment(*varDecl);
 		}
 		t("returns", retParams.empty() ? "" : " -> " + retParams);
+
+		string const& additionalParams = _function.isConstructor() ? m_additionalParameters[&_function] : "";
+		t("additionalParams", additionalParams);
+		t("comma", params.empty() || additionalParams.empty() ? "" : ", ");
+		t("additionalConstructorCode", generateAdditionalConstructorBody(_function));
```

Comment:
Also since constructors do not have return variables, functions and constructors are rather different - maybe just have the code duplication with regards to parameters and use two completely different templates?

Also I don't like the fact that we inject code here that is generated in a different function.

Reply:
Ok, I'll move constructor case to separate function than